api/shared/src/main/scala/dagr/api/DagrApi.scala

```diff
@@ -0,0 +1,130 @@
+  object Keys {
+    val ApiVersion = "dagr.api.version"
+    val ApiRoot    = "dagr.api.root"
+  }
+}
+*/
+
+/*
+trait DagrApiConfiguration extends Configuration {
+  lazy val version: String = optionallyConfigure[String](DagrApiConfiguration.Keys.ApiVersion).getOrElse("v1")
+  lazy val root: String    = optionallyConfigure[String](DagrApiConfiguration.Keys.ApiRoot).getOrElse("api")
+}
+*/
+
+trait DagrApiConfiguration {
+  lazy val version: String = "v1"
+  lazy val root: String    = "api"
+}
+
+object DagrApi {
+  type TaskId = BigInt
+
+  // *** IMPORTANT *** this must be the fully-qualified path to the DagrApi trait!
+  val prefixSegments: List[String] = List("dagr", "api", "DagrApi")
+}
+
+// NB: this trait cannot extend anything, as required by autowire
+trait DagrApi {
+
+
+  final def query(name: Seq[String] = Seq.empty, // any name pattern
+                  status: Seq[String] = Seq.empty, // name
+                  id: Seq[TaskId] = Seq.empty, // any task ids
+                  attempts: Seq[Int] = Seq.empty, // any attempt #
+                  minResources: Option[String] = None, // TODO: why cant this be of type ResourceSet
+                  maxResources: Option[String] = None, // TODO: why cant this be of type ResourceSet
+                  since: Option[String] = None, // TODO: why cant this be of type Instant
+                  until: Option[String] = None, // TODO: why cant this be of type Instant
+                  dependsOn: Seq[TaskId] = Seq.empty,
+                  dependents: Seq[TaskId] = Seq.empty,
+                  parent: Option[TaskId] = None, // TODO: include this in the response
+                  children: Seq[TaskId] = Seq.empty,
+                  script: Boolean = false,
+                  log: Boolean = false,
+                  report: Boolean = false
+                 ): TaskInfoResponse = {
+
+    def toResourceSet(str: String): ResourceSet = {
+      str.split(",").toList match {
+        case cores :: memory :: Nil => new ResourceSet(Cores(cores.toDouble), Memory(memory))
+        case _ => throw upickle.Invalid.Json(str, s"Could not extract resource set '$str'")
+      }
+    }
+
+    _query(
+      name         = name,
+      status       = status,
+      id           = id,
+      attempts     = attempts,
+      minResources = minResources.map(toResourceSet),
+      maxResources = maxResources.map(toResourceSet),
+      since        = since,
+      until        = until,
+      dependsOn    = dependsOn,
+      dependents   = dependents,
+      parent       = parent,
+      children     = children,
+      script       = script,
+      log          = log,
+      report       = report
+    )
+  }
+
+  def log(id: TaskId): String
+
+  def script(id: TaskId): String
+
+  def report(id: TaskId): String = "Not yet implemented"
+
+  def statuses(): Seq[TaskStatus]
+
+  /** The method a server implementation should implement. */
+  protected def _query(name: Seq[String] = Seq.empty, // any name pattern
```

Comment:
Only here so we can convert `String` to `ResourceSet`.  I don't think this is necessary anymore since I moved `ResourceSet` into the `dagr.api` package.